### PR TITLE
fix: mergefix origin/release

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Cherry-picks and mergefixes changes introduced by https://github.com/shapeshift/web/pull/8348 (present in [release](https://github.com/shapeshift/web/pull/8337 only, but not develop) on top of https://github.com/shapeshift/web/pull/8350 and https://github.com/shapeshift/web/pull/8309 (present in develop only), so we can run the release script.

This also fixes one regression in https://github.com/shapeshift/web/pull/8362/commits/895e0c00fa61c1f3154b68b72cb301a7feedf76b, introduced by https://github.com/shapeshift/web/pull/8309 i.e present in develop but not in release re: failing approvals for some swappers

image

## Issue (if applicable)

closes N/A

## Risk

> High Risk PRs Require 2 approvals

> What protocols, transaction types, wallets or contract interactions might be affected by this PR?

Low to none, ensure mergefix looks sane

## Testing

- Sanity test you can still execute Li.Fi bridges
- Sanity test you can still execute Li.Fi same-chain swaps

### Engineering

- ^

### Operations

- [ ] :checkered_flag: My feature is behind a flag and doesn't require operations testing (yet)

- ^

## Screenshots (if applicable)

- Li.Fi approvals and swap e2e

https://jam.dev/c/4603d1a2-b00d-4a58-810c-7c5d2c18595b

- Li.Fi approvals and bridge e2e

https://jam.dev/c/911d5019-e0d3-433c-ab7f-91c75b52fc73

- ZRX approvals and swap e2e

https://jam.dev/c/f29c2929-b05e-4033-a096-f970e4ebbd32

- CoW approvals and swap e2e

https://jam.dev/c/cab5446a-c9e1-4fd6-a526-977e74095e5a

- Arb Bridge withdraw approvals and bridge e2e (no approval needed for ARB bridge withdraws as this feature is part of the protocol)

https://jam.dev/c/63f6bbea-3846-4f0c-acc1-a484c9faa4c4

- Arb Bridge deposit approvals and bridge e2e

https://jam.dev/c/3e5780d5-e0ef-4cde-bbab-ece8d4285134

- Chainflip swap approvals and bridge e2e (no approval needed as Chainflip swaps are sends, contract calls)

https://jam.dev/c/8c82a6bf-59e0-43d6-aa24-1833bc3497bb

### CODE REVIEW

File: packages/swapper/src/swappers/LifiSwapper/getTradeQuote/getTradeQuote.ts [hunk 1]
```diff
@@ -259,11 +268,28 @@ export async function getTrade({
...
+        .map(current => current.tool)
+
+      const lifiExchangeTools = selectedLifiRoute.steps
+        .filter(
+          current =>
+            // A step tool is an exchange (swap) tool if all of its steps are non-cross-chain steps
+            current.includedSteps?.every(includedStep => includedStep.type !== 'cross'),
+        )
+        .map(current => current.tool)
+
```
Comment (gomesalexandre) [inline packages/swapper/src/swappers/LifiSwapper/getTradeQuote/getTradeQuote.ts:284]: This is already in release

File: packages/swapper/src/swappers/LifiSwapper/getTradeQuote/getTradeQuote.ts [hunk 2]
```diff
@@ -259,11 +268,28 @@ export async function getTrade({
...
       return {
         id: selectedLifiRoute.id,
         receiveAddress,
         quoteOrRate,
-        lifiTools: selectedLifiRoute.steps.map(step => step.tool),
+        lifiTools: {
+          bridges: lifiBridgeTools.length ? lifiBridgeTools : undefined,
+          exchanges: lifiExchangeTools.length ? lifiExchangeTools : undefined,
+        },
```
Comment (gomesalexandre) [inline packages/swapper/src/swappers/LifiSwapper/getTradeQuote/getTradeQuote.ts:292]: This is already in release

File: packages/swapper/src/swappers/LifiSwapper/getTradeQuote/getTradeQuote.ts [hunk 3]
```diff
@@ -315,6 +341,7 @@ export const getTradeQuote = async (
   lifiChainMap: Map<ChainId, ChainKey>,
 ): Promise<Result<LifiTradeQuote[], SwapErrorRight>> => {
   const quotesResult = await getTrade({ input, deps, lifiChainMap })
+
```
Comment (gomesalexandre) [inline packages/swapper/src/swappers/LifiSwapper/getTradeQuote/getTradeQuote.ts:344]: This is already in release

File: packages/swapper/src/swappers/LifiSwapper/utils/types.ts
Comment (gomesalexandre) [inline packages/swapper/src/swappers/LifiSwapper/utils/types.ts:1]: This is already in release

File: packages/swapper/src/swappers/ArbitrumBridgeSwapper/getTradeRate/getTradeRate.ts [hunk 4]
```diff
@@ -82,7 +83,7 @@ export async function getTradeRate(
           rate,
           buyAsset,
           sellAsset,
-          accountNumber: undefined,
+          accountNumber,
```
Comment (gomesalexandre) [inline packages/swapper/src/swappers/ArbitrumBridgeSwapper/getTradeRate/getTradeRate.ts:86]: This was missing in the step and was causing some failing approvals

**Review by NeOMakinG — APPROVED** [review]
https://jam.dev/c/cc964c86-984c-4b34-a820-78c671b2dc0a

Get this in so we can handle the release

Li.Fi bridges are taking a long time or failing at fetching the TX status, not related to those changes thought

**Pull request merged**